src/components/input/input.js

```diff
@@ -524,7 +524,13 @@ function mdMaxlengthDirective($animate, $mdUtil) {
       element.on('input keydown keyup', function() {
```

Comment:
I believe you could achieve the same without using a watcher by doing `ngModelCtrl.$parsers.push(renderCharCount);` here. Also it would be nice if you could add a unit test for this so it doesn't get broken in the future.